gcp/modules/gcp-stackdriver-alerting/client.rb

```diff
@@ -0,0 +1,172 @@
+require "json"
+require "google/cloud/monitoring"
+
+@project_id = ENV['PROJECT_ID']
+@destroy_triggered = ENV['DESTROY']
+@debug_mode = ENV['DEBUG']
+@resource_dir="#{File.expand_path(File.dirname(__FILE__))}/resources_rendered"
+
+def get_notification_channel_attribute(notification_channel)
+  if notification_channel["labels"]["email_address"]
+    result = notification_channel["labels"]["email_address"]
+  end
+
+  return result
+end
+
+def get_uptime_check_attribute(uptime_check)
+  if uptime_check["display_name"]
+    result = uptime_check["display_name"]
+  end
+
+  return result
+end
+
+def get_alert_policy_attribute(alert_policy)
+  if alert_policy["display_name"]
+    result = alert_policy["display_name"]
+  end
+
+  return result
+end
+
+def process_notification_channels(notification_channels = [])
+  notification_channel_service_client = Google::Cloud::Monitoring::NotificationChannel.new(version: :v3)
+  formatted_parent = Google::Cloud::Monitoring::V3::NotificationChannelServiceClient.project_path(@project_id)
+
+  stackdriver_notification_channels = {}
+  processed_notification_channels = {}
+
+  notification_channel_service_client.list_notification_channels(formatted_parent).each do |notification_channel|
+    puts notification_channel.to_json if @debug_mode
+    stackdriver_notification_channels[get_notification_channel_attribute(notification_channel)] = notification_channel
+  end
+
+  notification_channels.each do |notification_channel|
+    notification_channel_attribute = get_notification_channel_attribute(notification_channel)
```

Comment:
I think I understand what you're trying to do with `attribute` here, but it's such an abstract name that I found it hard to follow the logic. How about replacing `attribute` with `name` or maybe `identifier`?

Reply:
These functions are needed because we can't rely on Stackdriver resource names and want to be able to identify resources by some other attribute / set of attributes. But yeah, I don't like `_attribute` postfix either. `_identifier` probably makes more sense.